integrations/gitlab/gitlab_integration/gitlab_service.py

```diff
@@ -119,9 +123,38 @@ async def _get_file_paths(
             file["path"]
             for file in files
             if (not return_files_only or file["type"] == "blob")
-            and does_pattern_apply(path, file["path"] or "")
         ]
 
+    async def get_paginated_file_paths(
+        self,
+        project: Project,
+        path: str | List[str],
+        commit_sha: str,
+        return_files_only: bool = False,
+    ) -> AsyncIterator[list[str]]:
+        if not isinstance(path, list):
+            path = [path]
+        try:
+            async for files in AsyncFetcher().paginate_repository_tree(
+                project=project,
+                filtering_callable=does_pattern_apply,
+                filtering_paths=path,
+                ref=commit_sha,
+                recursive=True,
+            ):
+                if files:
+                    yield [
+                        file["path"]
+                        for file in files
+                        if (not return_files_only or file["type"] == "blob")
+                    ]
```

Comment:
worth checking if we can return the size as well to reduce redundant queries for files that are larger than allowed size

Reply:
We can't sadly